api/lib/opentelemetry/metrics/measure.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Metrics
+    # Measure is a contract between the library exposing the raw measurement and SDK aggregating these values into the
+    # @see Metric. Measure is constructed from the @see Meter class by providing set of Measure identifiers.
+    class Measure
+      NOOP_MEASUREMENT = :__noop_measurement__
+
+      private_constant(:NOOP_MEASUREMENT)
+
+      def create_measurement(value)
+        raise ArgumentError if value.negative?
+
+        NOOP_MEASUREMENT
+      end
+    end
+
+    # TODO: Do we want the type-specific class?
+    # DoubleMeasure is a Measure creating double measurements.
+    class DoubleMeasure < Measure
+      def create_measurement(value)
+        raise ArgumentError if value.integer?
```

Comment:
I think we should drop this check. It should be legal to pass an integer, as long as it is within the double-precision floating point range.